src/iarc7_motion/iarc_tasks/hold_position_task.py

```diff
@@ -0,0 +1,160 @@
+            self._K_X = rospy.get_param('~k_position_z')
+            self._K_Y = rospy.get_param('~k_position_z')
+            self._K_Z = rospy.get_param('~k_position_z')
+        except KeyError as e:
+            rospy.logerr('Could not lookup a parameter for hold position task')
+            raise
+
+        self._state = HoldPostionTaskStates.init
+
+    def _current_velocity_callback(self, data):
+        with self._lock:
+            self._drone_odometry = data
+
+    def get_desired_command(self):
+        with self._lock:
+            if (self._state != HoldPostionTaskStates.holding):
+                if self._drone_odometry is None:
+                    self._state = HoldPostionTaskStates.waiting
+                    return (TaskRunning(), NopCommand())
+                else:
+                    self._set_targets()
+                    self._state = HoldPostionTaskStates.holding
+            elif not (self._height_checker.above_min_maneuver_height(
+                        self._drone_odometry.pose.pose.position.z)):
+                return (TaskAborted(msg='Drone is too low'),)
+            elif not (self._check_max_error()):
+                return (TaskAborted(msg='Position holder is not for translating'),)
+
+            if self._canceled:
+                return (TaskCanceled(),)
+
+            if not self._check_max_start_hold_range():
+                return (TaskAborted(msg='The provided hold position is too far'),)
+
+            # p-controller
+            x_vel_target = ((self._x_position - self._drone_odometry.pose.pose.position.x)
+                                * self._K_X)
+            y_vel_target = ((self._y_position - self._drone_odometry.pose.pose.position.y) 
+                                * self._K_Y)
+            z_vel_target = ((self._z_position - self._drone_odometry.pose.pose.position.z) 
+                                * self._K_Z)
+
+            #caps velocity
+            vel_target = math.sqrt(x_vel_target**2 + y_vel_target**2)
+
+            if vel_target > self._MAX_TRANSLATION_SPEED:
+                x_vel_target = x_vel_target * (self._MAX_TRANSLATION_SPEED/vel_target)
+                y_vel_target = y_vel_target * (self._MAX_TRANSLATION_SPEED/vel_target)
+            
+            if (abs(z_vel_target) > self._MAX_Z_VELOCITY):
+                z_vel_target = z_vel_target/abs(z_vel_target) * self._MAX_Z_VELOCITY
+
+            velocity = TwistStamped()
+            velocity.header.frame_id = 'level_quad'
+            velocity.header.stamp = rospy.Time.now()
+            velocity.twist.linear.x = x_vel_target
+            velocity.twist.linear.y = y_vel_target
+            velocity.twist.linear.z = z_vel_target
+            
+            return (TaskRunning(), VelocityCommand(velocity)) 
+            
+    def cancel(self):
+        rospy.loginfo('HoldPosition Task canceled')
+        self._canceled = True
+
+    def _check_max_error(self):
```

Comment:
This function is exactly the same as _check_max_start_hold_range. We don't need two functions to do the same thing.